src/compile.c

```diff
@@ -560,10 +560,12 @@ dcc_build_somewhere(char *argv[],
         goto run_local;
     }
 
-    /* Lock the local CPU, since we're going to be doing preprocessing
-     * or include scanning. */
-    if ((ret = dcc_lock_local_cpp(&local_cpu_lock_fd)) != 0) {
-        goto fallback;
+    if (!dcc_is_preprocessed(input_fname)) {
+        /* Lock the local CPU, since we're going to be doing preprocessing
+         * or include scanning. */
+        if ((ret = dcc_lock_local_cpp(&local_cpu_lock_fd)) != 0) {
```

Comment:
`dcc_unlock(local_cpu_lock_fd);`
below this will fail when using pump mode, spewing unnecessary errors. That unlock needs to be conditioned on `local_cpu_lock_fd != -1` like the others at 
`unlock_and_clean_up:`